SimPEG/electromagnetics/frequency_domain/survey.py

```diff
@@ -97,3 +106,34 @@ def get_sources_by_frequency(self, frequency):
             frequency in self._frequency_dict
         ), "The requested frequency is not in this survey."
         return self._frequency_dict[frequency]
+
+    @property
+    def source_location_by_sounding_dict(self):
+        """
+        Source locations in the survey as a dictionary
+        """
+        return self._source_location_by_sounding_dict
+
+    def get_sources_by_sounding_number(self, i_sounding):
+        """
+        Returns the sources associated with a specific source location.
+        :param float i_sounding: source location number
+        :rtype: dictionary
+        :return: sources at the sepcified source location
+        """
+        assert (
+            i_sounding in self._source_location_dict
+        ), "The requested sounding is not in this survey."
+        return self._source_location_dict[i_sounding]
+
+    @property
+    def vnD_by_sounding_dict(self):
```

Comment:
Same here, type it and remove from name.
```suggestion
    def vnD_by_sounding(self) -> dict:
```